### PR TITLE
IGAPP-730: Report bad cms responses to sentry

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: web/src/components/FailureSwitcher.tsx [hunk 1]
```diff
@@ -13,6 +22,12 @@ type PropsType = {
 const FailureSwitcher = ({ error }: PropsType): ReactElement => {
   const { t } = useTranslation('error')
 
+  useEffect(() => {
+    if (!(error instanceof NotFoundError) && !(error instanceof FetchError)) {
```
Comment (f1sh1918) [inline web/src/components/FailureSwitcher.tsx:26]: why we can not use your created custom hook? useReportsError

Comment (f1sh1918) [reply web/src/components/FailureSwitcher.tsx:26]:
```suggestion
   useReportError(error)
```

**Review by f1sh1918 — APPROVED** [review]

File: web/src/components/FailureSwitcher.tsx (hunk 1 above)
Comment (klinzo) [reply web/src/components/FailureSwitcher.tsx:26]: I only added that hook for native as it is only used once in web.

File: native/src/sagas/watchFetchCities.ts [hunk 2]
```diff
@@ -16,6 +17,7 @@ export function* fetchCities(dataContainer: DataContainer, action: FetchCitiesAc
     yield* put(insert)
   } catch (e) {
     console.error(e)
+    reportError(e)
```
Comment (Znippet) [inline native/src/sagas/watchFetchCities.ts:20]: As I for my part don't like the console.errors in the unit tests much perhaps we could use the new reportError hook for this also and make it configurable?

I had a quick look, seems there are mor places where we have a console.error and the new reportError Hook at the same time.

Locally (developing) we don't need the Sentry report and on the other hand the console.errors aren't used for Sentry supervision.

Only an idea - and if we do it probalby a new task would make sense

**Review by Znippet — APPROVED** [review]

File: native/src/sagas/watchFetchCities.ts (hunk 2 above)
Comment (klinzo) [reply native/src/sagas/watchFetchCities.ts:20]: Sure, could you create a task? We could also suppress console errors in unit tests by replacing it with a mock I guess.

**Pull request merged**